core/shared/lib/src/logger/logger_util.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:logger/logger.dart';
+
+import 'abstract_logger.dart';
+
+enum LoggerLevel {
+  verbose,
+  debug,
+  info,
+  warning,
+  error,
+  fatal,
+}
+
+// Concrete implementation of AbstractLogger using the Logger library
+// Should the logger library change or be replaced,
+// then only this class needs to change, while its usages would stay the same
+class LoggerUtil extends AbstractLogger {
+
+  var logger = Logger();
```

Comment:
```suggestion
  final Logger logger;
  
    LoggerUtil({required this.logger});
```

We should pass the external dependencies in the constructor

Reply:
And then register the external dependency and the LoggerUtil inside the shared_injection_module
```suggestion
    InjectionModule.injector.registerLazySingleton<AbstractLogger>(
        () => LoggerUtil(logger: InjectionModule.injector()));
    InjectionModule.injector.registerLazySingleton<Logger>(() => Logger());
```